### DISCUSSION TITLE
Retrieving Unit IDs via API

### DISCUSSION THREAD

**Discussion by fenixero** [WeblateOrg/weblate#13146] [Q&A]
I am creating new entries for a component through the API:
`POST /api/translations/(string: project)/(string: component)/(string: language)/units/`
This endpoint allows me to create new entries in the component's only for default language.
How can I post translations for a key without knowing the unit ID (for the other languages)?
Or how can I retrieve the IDs of all units associated with a key?

**Comment by nijel** [comment]
There is currently no direct way for this. Let's track this as https://github.com/WeblateOrg/weblate/issues/13186

As a workaround, you can use upload with a file containing the single string only. It will match by key which you've chosen when creating the string.
